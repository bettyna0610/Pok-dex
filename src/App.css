.App {
  text-align: center;
}

.pagination {
  margin-top:30px;
}
.dropdown-menu{
  height:200px;
  overflow-y:auto;
  width:fit-content
}

.card{
  border-radius: 4px;
  background: #fff;
  box-shadow: 0 6px 10px rgba(0,0,0,.08), 0 0 6px rgba(0,0,0,.05);
    transition: .3s transform cubic-bezier(.155,1.105,.295,1.12),.3s box-shadow,.3s -webkit-transform cubic-bezier(.155,1.105,.295,1.12);
    
cursor: pointer;
}

.card:hover{
   transform: scale(1.05);
box-shadow: 0 10px 20px rgba(0,0,0,.12), 0 4px 8px rgba(0,0,0,.06);
}
.page-item-color {
  background-color: #d9534f !important;
}

.input-group.md-form.form-sm.form-1 input {
  border: 1px solid #bdbdbd;
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
  max-width:300px
}
.card-title {
  color:black
}