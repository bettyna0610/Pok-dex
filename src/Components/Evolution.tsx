import React, {useState} from 'react'

export const Evolution :React.FC<{name:string}> = (props) => {
    let [imageSource, setImage] = useState<any>('')
     
    
        fetch(`https://pokeapi.co/api/v2/pokemon/${props.name}`).then(response => response.json())
        .then(data => {
            if(data.sprites.other.dream_world.front_default) {
                setImage(data.sprites.other.dream_world.front_default)
            } else if (data.sprites.other["official-artwork"].front_default) {
                 setImage(data.sprites.other["official-artwork"].front_default)
         } else {
             setImage(data.sprites.front_default)
         }
            
        })
        
    

    return (
        <div>
            <div>{props.name}</div>
            <img src={imageSource}/>
        </div>
    )
}