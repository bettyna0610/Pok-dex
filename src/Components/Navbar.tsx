import React from 'react'
import {Navbar} from 'react-bootstrap'
import { Button } from 'react-bootstrap';


export const NavBarEl = () => {
    return  <Navbar  bg="dark" expand="lg">
        <Navbar.Brand >Catch them all!</Navbar.Brand>
        
    </Navbar>
    


}