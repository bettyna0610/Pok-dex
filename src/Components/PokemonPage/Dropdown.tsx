import React from 'react'
import DropdownButton from 'react-bootstrap/DropdownButton';
import Dropdown from 'react-bootstrap/Dropdown'



export const DropdownList : React.FC <{moves:string[]}> = (props) => {
   return (
   <DropdownButton
      alignRight
      title="Moves"
      variant="danger"
        >
            {props.moves.map((move:string) =>  <Dropdown.ItemText key={move} >{move}</Dropdown.ItemText> )}
                     
    </DropdownButton>
   )
};
