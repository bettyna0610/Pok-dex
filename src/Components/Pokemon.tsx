import React from 'react'
import {Evolution} from './Evolution'

type MyProps = {
    match:any
}

type MyState = {
 
}

export class Pokemon extends React.Component<MyProps, MyState> {
   
    state = {
        image:'',
        name:'',
        abilities: [],
        type:[],
        orderNumber:0,
        stats: {
            hp:'',
            attack:'',
            defense:'',
            specialAttack:'',
            specialDefense:'',
            speed:''
        },
        evolution: []
        
            
        ,
        moves: []

    }

    componentDidMount() {
        let {pokemonIndex}  = this.props.match.params
        fetch(`https://pokeapi.co/api/v2/pokemon/${pokemonIndex}`).then(response => response.json())
        .then(data => { 
            let imageSource;
            if(parseInt(pokemonIndex) >= 650) {
             imageSource = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemonIndex}.png`
            } else {
                imageSource =  `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemonIndex}.svg`
            }
            
                let abilities = data.abilities.map((ability:any) => ability.ability.name)
                let name = data.name
                let type = data.types.map((pokType:any) => pokType.type.name)
                let orderNumber = data.order
                let hp; let attack; let defense; let specialAttack; let specialDefense; let speed;
                let moves = data.moves.map((move:any) => <li>{move.move.name}</li>)

                data.stats.map(
                   (stat:any)  => {
                       switch(stat.stat.name) {
                           case 'hp':
                               hp = stat['base_stat']
                               break;
                         case 'attack':
                             attack = stat['base_stat']
                             break;
                             case 'defense':
                                 defense = stat['base_stat']
                                 break;
                                 case 'special-attack':
                                     specialAttack = stat['base_stat']
                                     break;
                                     case 'special-defense':
                                         specialDefense = stat['base_stat']
                                         break;
                                         case 'speed':
                                             speed = stat['base_stat']
                                     break;
                       } 


                    
                })
                
                fetch(data.species.url).then(response => response.json())
               .then(dataEvolution => { fetch(dataEvolution.evolution_chain.url).then(response => response.json()).then(dataEvoChain => {
                   let evo1  ; let evo2; let evo3; let evolutionArray = []
                   
                    evo1 = dataEvoChain.chain.species.name;
                   
                         
                    if(dataEvoChain.chain.evolves_to.length > 0)
                    {
                      if(dataEvoChain.chain.evolves_to[0].species.name.length > 0) {

                    evo2 = dataEvoChain.chain.evolves_to[0].species.name;

                      //most tettem ide
                    if(dataEvoChain.chain.evolves_to[0].evolves_to.length > 0) {
                     evo3 = dataEvoChain.chain.evolves_to[0].evolves_to[0].species.name
                    } else {
                        evo3 = null
                    }

                   } else {
                       evo2 = null
                     }
                   }
                   evolutionArray.push(evo1,evo2,evo3)
                  
                   this.setState({
                    evolution: evolutionArray
                })})})

                   

               this.setState({
                    image: imageSource,
                    abilities,
                    name,
                    type,
                    orderNumber,
                    stats: {
                        hp,
                        attack,
                        defense,
                        specialAttack,
                        specialDefense,
                        speed
                    },
                    moves

               })

               
            })
           
               

            
        
    }


render() {
    
    let imgSource:string;
    let {image,name,abilities,type,orderNumber,stats,moves,evolution} = this.state
    fetch(`https://pokeapi.co/api/v2/pokemon/${name}`).then(response => response.json())
    .then(data => {
        imgSource = data.sprites.front_default
        console.log(imgSource)
    })
    
    return (
        <div><img src={image}/>
       <div>
       {name}
           </div>
         {abilities}
         <div>{type}</div>
         <div>{orderNumber}</div>
         {stats.hp}<div>{stats.attack}</div><div>{stats.defense}
         </div><div>{stats.specialAttack}</div><div>{stats.specialDefense}</div><div>{stats.speed}</div>
         Moves:<ul>{moves}</ul>
          {/*<div>{evolution.evo1}</div><div>{evolution.evo2 && evolution.evo2}</div><div>{evolution.evo3 && evolution.evo3}</div>*/}
         {evolution.map( ( evo:string) => <Evolution name={evo} source={imgSource}/>)}
         </div>
    )
   
}
}
