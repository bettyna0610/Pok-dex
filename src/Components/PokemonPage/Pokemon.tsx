
import React from 'react'
import {DropdownList} from './Dropdown'
import {Link} from 'react-router-dom'
import {Stats} from './Stats'
import {Evolution} from './Evolution'

type MyProps = {
    match:any
}

type MyState = {
 abilities: string[],
 image:string,
 name:string,
 type: string [],
 orderNumber:number,
 moves: string [],
 loading: boolean,
 dataPokemon: any
}


export class Pokemon extends React.Component<MyProps, MyState> {
   
    state = {
        image:'',
        name:'',
        abilities: [],
        type:[],
        orderNumber:0,
        moves: [],
        loading:false,
        dataPokemon: []
    }

    async componentDidMount() {

        let pokemonIndex  = this.props.match.url.split('/')[this.props.match.url.split('/').length-1]
        let image;
        this.setState({
          loading: true
        })
        const response = await fetch(`https://pokeapi.co/api/v2/pokemon/${pokemonIndex}`)
        const data =  await response.json();

        this.setState({
          dataPokemon: data
        })

            if(data.sprites.other.dream_world.front_default) {
             image = data.sprites.other.dream_world.front_default
            } else if (data.sprites.other["official-artwork"].front_default) {
             image= data.sprites.other["official-artwork"].front_default
            } else {
             image = data.sprites.front_default
            }
    
            
            let abilities :string[] = data.abilities.map((ability:any) => ability.ability.name)
            let name = data.name
            let type = data.types.map((pokType:any) => pokType.type.name)
            let orderNumber = data.order
            let moves = data.moves.map((move:any) => move.move.name)
          
               
               this.setState({
                image,
                abilities,
                name,
                type,
                orderNumber,
                moves,
                loading:false            
              })   
    }


render() {
    
     let {image,name,abilities,type,orderNumber,moves} = this.state
     
     
    return (

    <div className="container">
      {this.state.loading  ? "Loading..." :
      <div className="col justify-content-center">
        <div>  
          <div className="row mt-3 mb-0">
            <div className="col-lg-4 col-sm-4 col-xs">
              <Link to='/' className="btn btn-danger">Back to Pok√©dex</Link>
            </div>
            <div className="col-lg-4 col-sm-4 col-xs center-block">
              <div className="row justify-content-center">{orderNumber}.</div>
              <div className="row justify-content-center ">
               <h2>{name.toLowerCase().split(" ").map(character => character.charAt(0).toUpperCase() + character.substring(1))}</h2>
              </div> 
            </div>
            <div className="col-4"></div>  
          </div>
      <hr></hr>
      <div className="row mt-1">
        <div className="col-lg-4 col-sm-4 align-items-center col-xs">
          <img width="200px" height="200px" className="img-fluid" src={image}/>
        </div>
        <div className="col-lg-4 col-sm-4 col-xs">
          <div className="row justify-content-center col-xs">
            <h5>Types:</h5>
          </div> 
          <div className="row justify-content-center">
            {type.map((type:string) => <div key={type} className="btn btn-danger m-2">{type}</div>)}
          </div>
        </div>
        <div className="col-lg-4 col-sm-4 align-items-left col-xs">
          <div className="row justify-content-center">
            <h5>Abilities:</h5> 
          </div>  
          <div className="col">
            {abilities.map((ability:string) => <div key={ability} className="btn btn-danger m-2">{ability}</div>)}
          <div>
        </div>
       </div>
          <div className="row mt-3 justify-content-center">
            <DropdownList moves={moves} />
          </div>
        </div>
      </div>
     <div>
    <div className="row justify-content-center">
      <div className="col-lg-3 col-sm-3"></div>
      <div className="col-lg-6 col-sm-6 col-xs">
        <div>
           <h5> Stats:</h5>
        </div>
          <Stats  data={this.state.dataPokemon}
           />
      </div>
      <div className="col-lg-3 col-sm-3"></div>
   </div>
  
    <div className="row justify-content-center m-3"> 
      <h5> Evolution:</h5> 
    </div>
    <div className="row justify-content-center m-3">
      <Evolution data={this.state.dataPokemon}/>
      
    </div> 
   </div>
  </div>
 </div>
}
</div>         
    )
   
}
}
