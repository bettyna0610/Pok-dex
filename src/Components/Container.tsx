import React from 'react';
import {Pokemon} from './Pokemon'

type MyProps = {
  // using `interface` is also ok
  
};

type MyState = {
    pokemonAll: Data[];
   
  };

  type Data = {
    name:string;
    url:string
  }

export class PokemonContainer extends React.Component<MyProps,MyState> {
    state: MyState = {
        // optional second annotation for better type inference
        pokemonAll: [],
        
      };
    /*
    componentDidMount() {
      fetch('https://pokeapi.co/api/v2/pokemon/?limit=60').then(response => response.json())
      .then(data => {
          console.log(data.results[0])
          let newData = data.results[0].name
       
      }
      )
      console.log("hello")
      this.setState({
        pokemanAll: "new"
    })
      console.log("hey")
    }*/

    componentDidMount () {
      fetch('https://pokeapi.co/api/v2/pokemon/?limit=100').then(response => response.json())
      .then(data => {
          console.log(data.results[0])
          let newData = data.results
          let pokemonArray = newData.map((poke :Data) => {return poke.name} )
      this.setState({
       // pokemonAll: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
       pokemonAll: newData
      })

    

    })
  }
    
    handleClick = () => {
        this.setState({
            pokemonAll: []
        })
    }

    render() {
      let pokemonArray = this.state.pokemonAll
        return (
            <div className="container">
           <div className="row">
           <div className="col-xs-3">
            {pokemonArray.map(pokemon => <Pokemon  key={pokemon.name} url={pokemon.url} name={pokemon.name} /> )}
            </div>
           </div>
          
           
            
           
            </div>
        )
    }
}