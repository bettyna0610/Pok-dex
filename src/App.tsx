import React from 'react';
import './App.css';
import {PokemonContainer} from './Components/Container'
import 'bootstrap/dist/css/bootstrap.min.css';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
import {Pokemon} from './Components/Pokemon'

function App() {
  return (
    <Router>
     <div className="App">
      <Switch>
      <Route exact path="/" component={PokemonContainer} />
              <Route exact path="/pokemon/:pokemonIndex" component={Pokemon} />
      <PokemonContainer  />
      </Switch>
     
    </div>
    </Router>
    
  );
}

export default App;
