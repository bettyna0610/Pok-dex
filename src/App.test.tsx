import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import ReactDOM from 'react-dom';
import {PokemonContainer} from './Components/Container';
import {Pokemon} from './Components/PokemonPage/Pokemon';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'
import {DropdownList} from './Components/PokemonPage/Dropdown'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import userEvent from '@testing-library/user-event'



test('renders the component', () => {
  const component = shallow(<App />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('should render all dropdown list with moves', () => {
  const component = shallow(<DropdownList moves={["attack","defense","fly"]} />);
  expect(toJSON(component)).toMatchSnapshot();
})

test('not showing pagination when search', () => {
  let value = "pikachu"
  const component = shallow(<PokemonContainer />);
  component.find('input').simulate("change", { target: {value}})
  expect(component.find('.pagination')).toHaveLength(0)
});

test('pokemon page navigates back to homepage', () => {
  const history = createMemoryHistory()
  render(
    <Router history={history}>
      <Pokemon match=""/>
    </Router>
  )

  const leftClick = { button: 0 }
  userEvent.click(screen.getByText(/Back to Pok√©dex/i), leftClick)
  expect(screen.getByText(/Type/i)).toBeInTheDocument()

})