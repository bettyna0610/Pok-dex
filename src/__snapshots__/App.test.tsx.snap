// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/pokemon/:pokemonIndex"
      />
      <PokemonContainer />
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`should render all dropdown list with moves 1`] = `
<DropdownButton
  alignRight={true}
  title="Moves"
  variant="danger"
>
  <DropdownItemText
    key="attack"
  >
    attack
  </DropdownItemText>
  <DropdownItemText
    key="defense"
  >
    defense
  </DropdownItemText>
  <DropdownItemText
    key="fly"
  >
    fly
  </DropdownItemText>
</DropdownButton>
`;
